### DISCUSSION TITLE
`pocketbase update` questions

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#2669] [Q&A] [answered]
@ganigeorgiev I'm looking at `pocketbase update` for updating https://pockethost.io instances when versions change. A few questions:

1. Does it create a backup first? If so, is it guaranteed behavior for every future release?
2. Can `pocketbase update` be run multiple times with no harm?
3. Is there any `--autoupdate` type flag existing or planned so users could just do `pocketbase serve --autoupdate` and never worry about running `pocketbase update` manually?

Thank you!

**Comment by ganigeorgiev** [comment]
> 1. Does it create a backup first? If so, is it guaranteed behavior for every future release?

Yes. By default it'll create a backup of the app `pb_data`. If you don't want the automatic backup, you can disable it with `--backup=0`.

> 2. Can `pocketbase update` be run multiple times with no harm?

Yes. The `update` command will first fetch from GitHub the latest release info and compare it against the current version. If your app is already using the latest version you'll get an appropiate console  message. Otherwise - it will proceed downloading the latest release, extract and replace your current binary and finally backup your `pb_data`. If the backup or one of the other steps fails, it should revert the executable replacement.

> 3. Is there any `--autoupdate` type flag existing or planned so users could just do `pocketbase serve --autoupdate` and never worry about running `pocketbase update` manually?

No, but there are plans sometime in the future to allow triggering the `update` command from the Admin UI when the prebuilt executable is used.

**Reply by ganigeorgiev** [reply]
One eventual issue that could occur in your case if you are planning to run the command often I guess is hitting the GitHub public API rate limits:

> For unauthenticated requests, the rate limit allows for up to **60 requests per hour**. Unauthenticated requests are associated with the originating IP address, and not the person making requests.

**Comment by benallfree** [comment]
Oops, I entirely misunderstood the purpose of this command. It updates the binary?

I thought `update` ran internal migrations, but I guess that was `upgrade` right? 

I am trying to ensure that users can safely bump PocketBase versions for their instances. Thoughts?

**Reply by ganigeorgiev** [reply, marked as answer]
> It updates the binary?

Yes.

> I thought update ran internal migrations, but I guess that was upgrade right?

No, `upgrade` was a temporary command to automate some of the v0.7.x -> v0.8+ `pb_data` structure changes.
To manually apply db migrations, you can run `migrate up`, but for most of the cases this is not needed as db migrations (including system ones) are automatically executed on `serve`.

> I am trying to ensure that users can safely bump PocketBase versions for their instances. Thoughts?

I'm not sure what to recommend. Note that the `update` command will always fetch the latest available executable. In most cases this is safe as we rarely have breaking web APIs changes, but it still could happen.
I guess your issue will be partially resolved once we add the update option in the Admin UI, but until then you can try to create a secure external endpoint that will trigger the `update` command for your users (note that you'll have to restart the app process too).

**Comment by benallfree** [comment]
In my case it would actually be harmful to allow PocketBase to `update` itself because PocketHost keeps binaries of every version already. Depending on what version the instance is using, a shared `pocketbase` binary of the matching version will be executed.

In fact, I'd probably want a way to disable `update` or at least not expose it to the admin.

I think this ticket is resolved because it sounds like system-level migrations are handled automatically when `serve` is launched. My original question was meant to be about migrations, not binary updates. Sorry for the confusion.
